src/main/scala/tech/cryptonomic/conseil/Lorre.scala

```diff
@@ -17,43 +18,67 @@ object Lorre extends App with LazyLogging {
 
   private val conf = ConfigFactory.load
   private val awaitTimeInSeconds = conf.getInt("dbAwaitTimeInSeconds")
-  private val sleepIntervalInSeconds = conf.getInt("lorre.sleepIntervalInSeconds")
-  private val feeUpdateInterval = conf.getInt("lorre.feeUpdateInterval")
+  val sleepIntervalInSeconds = conf.getInt("lorre.sleepIntervalInSeconds")
 
   lazy val db = DatabaseUtil.db
   val tezosNodeOperator = new TezosNodeOperator(TezosNodeInterface)
 
-  var iterationsOfLorre = 0
-  var shouldFeeBeCalculatedThisIteration = true
+  val usage =
+    """
+      Usage: runMain tech.cryptonomic.conseil.Lorre --platform <platform> --network <network>
+    """.stripMargin
 
-  try {
-    while(true) {
-      processTezosBlocks()
-      processTezosAccounts()
-      if (shouldFeeBeCalculatedThisIteration) {
-        FeeOperations.processTezosAverageFees()
-      }
-      logger.info("Taking a nap")
-      iterationsOfLorre = iterationsOfLorre + 1
-      shouldFeeBeCalculatedThisIteration = iterationsOfLorre % feeUpdateInterval == 0
-      Thread.sleep(sleepIntervalInSeconds * 1000)
-    }
-  } finally db.close()
+  val argsList = args.toList
+  val (platform, network) = argsList match {
+    case "--platform" :: pl :: "--network"  :: nw :: Nil => (pl, nw)
+    case "--network"  :: nw :: "--platform" :: pl :: Nil => (pl, nw)
+    case _ =>
+      println(usage)
+      sys.exit(1)
+  }
+
+  val platformsAndNetworks = ConfigUtil.parsePlatformsAndNetworks()
+
+  val supportedPlatformsAndNetworks =
+    """
+      | The supported platforms and their associated networks are as follows:
+      | (Format: <platform>: <network_one>, <network_two>...)
+    """
+    .stripMargin + ConfigUtil.outputPlatformsAndNetworks.stripMargin
+
+  val networkPlatformValidation = conf.hasPath(s"platforms.$platform.$network")
+
+  networkPlatformValidation match {
```

Comment:
it's not that relevant to me, but pattern matching on bools can be frowned upon and it's not optimized as well as a simple`if-else`

Usually you should prefer

```
if (networkPlatformValidation) // or better networkPlatformValid (!)
  try {
    while (....
  }
else {
  println(...)
  system.exit(1)
}
```

Reply:
That's so counterintuitive! I thought pattern matching was to way to go always. Interesting

Reply:
In fact this is a scala thing... we were even thinking of asking for a compiler rewrite rule to convert it under the hood...

I've seen the pattern used in F# just the other day... I wonder what haskell people think